### ISSUE TITLE
can't successfully save file for SM-G970 

### ISSUE THREAD

**Issue by dchmelik** [ivanmeler/SamFirm_Reborn#5]
It's great SamFirm Reborn works for SM-T820 & SM-J727U, but SM-G970U (or G970U1) had problems.
&nbsp;&nbsp;&nbsp; It was bought direct from Samsung as unlocked (false advertising: only carrier-unlocked, not bootloader-unlocked) SM-G970U which Samsung.com calls SM-G970UZBATMB apparently last three letters being region code.  However, internally (*#1234#) it's G970U1 (sometimes different than G970U: ) specifically G970U1OYM6HVA1 you said longer string has region code but I couldn't find any matching.
&nbsp;&nbsp;&nbsp; Entering G970U & TMB in SamFirm Reborn downloaded then said couldn't decrypt: wrong key, then said finished anyway.  However, saved .zip.enc4 is unreadable by Windows & 7z: maybe didn't get new key to finish... should I download again or is there a log I can upload to potentially help fix?

**Comment by ivanmeler** [comment]
You likely wncountered network error while downloading did you have check crc32 checkbox enabled?

**Comment by dchmelik** [comment]
Yes: 'check CRC32' is checked/ticked when starting so I left it and all other defaults, but I'll try again.

**Comment by dchmelik** [comment]
Yes: 'check CRC32' is checked/ticked when starting so I left it and all other defaults, but tried again.
        It's almost 6GB so an overnight download, but keeps happening.  Before end it does say the below.

```
Checking CRC32...
Success: CRC match!

```
then...

```
Decrypting firmware...
Error decryptinf file: wrong key.
Decryption finished.
```

**Comment by ksmith3036** [comment]
I forked this project, and changed the authentication scheme (amongst other changes) used for the Samsung servers. I also fixed a bug for the resume download function, where the bug was caused by a 2 GByte limit in the http Range header in the .NET framework.
I tried downloading the mentioned combination of SM-G970U and TMB region, and it downloads and decrypts fine. If you still have the same issue, maybe you would like to test the fork: https://github.com/ksmith3036/SamFirm_Reborn
